### PR TITLE
Refactor(plugins): Move jinja filter code for arista.avd.hide_passwords to PyAVD

### LINKED ISSUE #4111: Refactor(plugins): Move jinja filter code for arista.avd.hide_passwords to PyAVD
### Enhancement summary

Move jinja filter code for arista.avd.hide_passwords to PyAVD

### Which component of AVD is impacted

plugins

### Use case example

please detail your use case

### Describe the solution you would like

The code and unit tests for hide_passwords filter should be moved to pyavd/j2filters and imported inside ansible plugins/filter using pyavd wrapper.

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Shivani-gslab**
## Change Summary

Move jinja filter code for arista.avd.hide_passwords to PyAVD

## Related Issue(s)

Fixes #4111 

## Component(s) name
Filter:

- `arista.avd.hide_passwords`

## Proposed changes
Moving the code and unit tests for arista.avd.list_compress filter to PyAVD

## How to test
CI should catch things.

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/tests/pyavd/j2filters/test_hide_passwords.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import pytest
+from pyavd.j2filters.hide_passwords import hide_passwords
+
+VLAID_INPUT_HIDE_PASSWORDS = [
```
Comment (laxmikantchintakindi) [inline python-avd/tests/pyavd/j2filters/test_hide_passwords.py]:
```suggestion
VALID_INPUT_HIDE_PASSWORDS = [
```

File: python-avd/tests/pyavd/j2filters/test_hide_passwords.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    (1233456, True, "<removed>"),
+    ({"key": "value"}, True, "<removed>"),
+    (None, True, None),
+]
+
+INVLAID_INPUT_HIDE_PASSWORDS = [("password", "wrong_type_flag", "wrong_type_flag in hide_passwords filter is not of type bool")]
+
+
+class TestHidePasswordsFilter:
+    @pytest.mark.parametrize("value, hide_passwords_flag, hidden_password", VLAID_INPUT_HIDE_PASSWORDS)
```
Comment (laxmikantchintakindi) [inline python-avd/tests/pyavd/j2filters/test_hide_passwords.py]:
```suggestion
    @pytest.mark.parametrize("value, hide_passwords_flag, hidden_password", VALID_INPUT_HIDE_PASSWORDS)
```

File: python-avd/tests/pyavd/j2filters/test_hide_passwords.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    def test_hide_passwords_valid(self, value, hide_passwords_flag, hidden_password):
+        """
+        Test hide_passwords
+        """
+        assert hide_passwords(value, hide_passwords_flag) == hidden_password
+
+    @pytest.mark.parametrize("value, hide_passwords_flag, error_msg", INVLAID_INPUT_HIDE_PASSWORDS)
```
Comment (laxmikantchintakindi) [inline python-avd/tests/pyavd/j2filters/test_hide_passwords.py]:
```suggestion
    @pytest.mark.parametrize("value, hide_passwords_flag, error_msg", INVALID_INPUT_HIDE_PASSWORDS)
```

File: python-avd/tests/pyavd/j2filters/test_hide_passwords.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+VLAID_INPUT_HIDE_PASSWORDS = [
+    ("dummy", False, "dummy"),
+    ("dummy", True, "<removed>"),
+    ([1, 2, 3], True, "<removed>"),
+    (1233456, True, "<removed>"),
+    ({"key": "value"}, True, "<removed>"),
+    (None, True, None),
+]
+
+INVLAID_INPUT_HIDE_PASSWORDS = [("password", "wrong_type_flag", "wrong_type_flag in hide_passwords filter is not of type bool")]
```
Comment (laxmikantchintakindi) [inline python-avd/tests/pyavd/j2filters/test_hide_passwords.py]:
```suggestion
INVALID_INPUT_HIDE_PASSWORDS = [("password", "wrong_type_flag", "wrong_type_flag in hide_passwords filter is not of type bool")]
```

File: python-avd/tests/pyavd/j2filters/test_hide_passwords.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+__metaclass__ = type
+
+import pytest
+from pyavd.j2filters.hide_passwords import hide_passwords
+
+VLAID_INPUT_HIDE_PASSWORDS = [
+    ("dummy", False, "dummy"),
+    ("dummy", True, "<removed>"),
+    ([1, 2, 3], True, "<removed>"),
```
Comment (laxmikantchintakindi) [inline python-avd/tests/pyavd/j2filters/test_hide_passwords.py]: Value is defined to be string and here list is written.

Comment (Shivani-gslab) [reply python-avd/tests/pyavd/j2filters/test_hide_passwords.py]: Thanks. 
The PR is not ready for review yet, I am still working on some issue.

File: python-avd/pyavd/j2filters/hide_passwords.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+
+def hide_passwords(value: list | dict | str | int | None, hide_passwords: bool = False) -> str:
+    """
+    hide_passwords replaces the input data by "<removed>" if the hide_passwords parameter is true
```
Comment (MaheshGSLAB) [inline python-avd/pyavd/j2filters/hide_passwords.py]: Replaces the input data with "<removed>" if the hide_passwords parameter is true.

File: python-avd/pyavd/j2filters/hide_passwords.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+
+def hide_passwords(value: list | dict | str | int | None, hide_passwords: bool = False) -> str:
```
Comment (MaheshGSLAB) [inline python-avd/pyavd/j2filters/hide_passwords.py]:
```suggestion
def hide_passwords(value: str, hide_passwords: bool = False) -> str:
```

File: python-avd/pyavd/j2filters/hide_passwords.py [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    """
+    hide_passwords replaces the input data by "<removed>" if the hide_passwords parameter is true
+
+    Args:
+        value (list | dict | str | int | None): Value to be hidden
```
Comment (MaheshGSLAB) [inline python-avd/pyavd/j2filters/hide_passwords.py]:
```suggestion
        value (str): Value to be hidden
```

File: python-avd/pyavd/j2filters/hide_passwords.py (hunk 6 above)
Comment (Shivani-gslab) [reply python-avd/pyavd/j2filters/hide_passwords.py]: according to the code, it replaces with `<removed>` string

Comment (MaheshGSLAB) [reply python-avd/pyavd/j2filters/hide_passwords.py:9]: yes sorry i mean the doc string should be like
Replaces the input data with "<removed>" if the hide_passwords parameter is true.

Comment (MaheshGSLAB) [reply python-avd/pyavd/j2filters/hide_passwords.py]: yes please use the same in double quote. Its getting removed while commenting

Comment (MaheshGSLAB) [reply python-avd/pyavd/j2filters/hide_passwords.py]:
```suggestion
    Replace the input data with "<removed>" if the hide_passwords parameter is true. 
```

**Review by Vibhu-gslab — APPROVED** [review]

File: python-avd/pyavd/j2filters/hide_passwords.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+    Args:
+        value (str): Value to be hidden
+        hide_passwords(bool) : Enable/disable hide_passwords
+    Returns:
+        str: "<removed>" or value
+
+    """
+    if not isinstance(hide_passwords, bool):
+        raise TypeError(f"{hide_passwords} in hide_passwords filter is not of type bool")
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/j2filters/hide_passwords.py:19]: IMO this check is not really needed. We perform a boolean evaluation of anything given. @gmuloc ?

**Review by ClausHolbechArista — APPROVED** [review]

File: python-avd/pyavd/j2filters/hide_passwords.py (hunk 9 above)
Comment (ClausHolbechArista) [reply python-avd/pyavd/j2filters/hide_passwords.py:19]: since we have tests and everything we can keep this for a later time.

**Pull request merged**